core/model/src/main/java/org/eclipse/sw360/antenna/model/util/ArtifactLicenseUtils.java

```diff
@@ -54,13 +57,19 @@ public static LicenseInformation getFinalLicenses(Artifact artifact) {
         final Optional<LicenseInformation> declared = artifact.askForGet(DeclaredLicenseInformation.class);
         final Optional<LicenseInformation> observed = artifact.askForGet(ObservedLicenseInformation.class);
         if(declared.isPresent() && observed.isPresent()) {
-            final LicenseStatement effective = new LicenseStatement();
-            effective.setLeftStatement(declared.get());
-            effective.setRightStatement(observed.get());
-            effective.setOp(LicenseOperator.AND);
-            return effective;
-        }
+            final String declaredLicenseRepresentation = declared.get().evaluate();
+            final String observedLicenseRepresentation = observed.get().evaluate();
 
+            if (observedLicenseRepresentation.contains(declaredLicenseRepresentation)) {
```

Comment:
Does `LicenseInformation::evaluate` sufficiently normalize the expression, so that the substring relation is correct? Or are there problems with non-deterministic ordering or unnecessary `(MIT AND MIT)` clauses?

Reply:
In the case where
- observed is `(MIT OR GPL)` and
- declared is `GPL`

I think the substring relation is incorrect. The correct output would be `GPL` or `(GPL AND (MIT OR GPL))`, but not `(MIT OR GPL)`.

Reply:
@maxhbr This is a crude simplification to get rid of some MIT and MIT cases, but in general the mathematical expression (Declared) AND (Observed) is right. There are cornercases which we cannot handle with the above approach like (MIT AND BSD-3) AND (BSD-3 AND MIT) but since we do not lose the mathematical correctness, this is fine.

The simplification is in order to reduce the amount of findings by project teams who stumble accross strange expressions. But I did not want to enter a mathematical expression solver that simplifies the expressions.

Concerning your point with MIT OR GPL. I do not think that this is a relevant case. I have no idea, how the observed license expression might contain an OR operator because the observed license is a summary of what a license scanner detected in the sources. A license scanner can only detect an OR if it detects a file that contains an OR expression and even that requires some understanding and semantics which are in general undefined. And to conclude finally that it is on component level only an OR expression means that there are only files in the codebase that contain this OR expression. So the project declares that it uses one license but in all files in the codebase there is either no expression or the OR expression. That is really a weirdo case.

But @bs-ondem to get rid of even that case, I propose we change it to: if the observed license expression contains an "OR" operator, we do not simplify the effective license to the observed license, instead we make the standard (Declared) AND (Observed).

Reply:
BTW. Can we express WITH in the Antenna data model?

Reply:
> But @bs-ondem to get rid of even that case, I propose we change it to: if the observed license expression contains an "OR" operator, we do not simplify the effective license to the observed license, instead we make the standard (Declared) AND (Observed).

Yes, that works

> BTW. Can we express WITH in the Antenna data model?

No, not yet.